### ISSUE TITLE
Allow Cupy to automatically detect CUDA 12 DLLs installed from PyPI Packages

### ISSUE THREAD

**Issue by martinResearch** [cupy/cupy#8720]
### Description

With NVIDIA now publishing Python wheels on PyPI that include essential CUDA 12 DLLs, installing CUDA dependencies for Cupy has become more accessible. By installing the following packages (or a subset thereof), users sould be able to obtain all the necessary DLLs required by Cupy:
* nvidia-nvjitlink-cu12
* nvidia-nvtx-cu12
* nvidia-cuda-runtime-cu12
* nvidia-cublas-cu12
* nvidia-cuda-cupti-cu12
* nvidia-cuda-nvrtc-cu12
* nvidia-cudnn-cu12
* nvidia-cufft-cu12
* nvidia-curand-cu12
* nvidia-cusolver-cu12
* nvidia-cusparse-cu12

**Current Issue:** Even with these packages installed, users still need to manually move or copy the header files and DLLs into common folders to make them discoverable by Cupy . This extra step complicates what could otherwise be a smooth, pip-compliant installation process.

**Feature Request:** It would be highly beneficial if Cupy could automatically locate the necessary header and DLL files from these PyPI-installed packages without requiring manual relocation. This enhancement would allow users to install Cupy and the required CUDA dependencies via a standard Python package installation without needing Conda or a manual CUDA installation.

**Proposed Solution:** To streamline the installation, Cupy could support an optional extra requirement, enabling users to simply run:
```
pip install cupy-cuda12x[cuda_dlls]
```
This command would install both Cupy and the relevant CUDA DLL packages in one step, improving the user experience by simplifying the installation process.

**Comment by kmaehashi** [comment]
Thanks for the feedback @martinResearch! This one is tracked in #8013, and we plan to support it in CuPy v14 (https://github.com/cupy/cupy/issues/8269).

**Comment by owoodhansen** [comment]
Dear @martinResearch 

It sounds like we are in a similar predicament, and while its great that this will be supported in newer cupy's im still working with the latest stable 13.3. After spending 1½ day on trying to make it cublas work in python, I finally found a stupid but functioning solution. I'll share it here, but im curious if you found a smarter way?

Sorry if im describing things in a basic way, but I tried my best to be clear:
Setup is Windows 10 without the full CUDA Toolkit. I only want to use wheels. Within site-packages\ of my python install, i have the following relevant folders: cupy; cupyx; cupy_backends (all three installed at once with cupy-cuda12x); nvidia-pyindex; and nvidia. 

The nvidia\ folder contains subfolders for each cuda-thing i have installed with wheels (cublas; curand; cuda_runtime etc). I set the CUDA_PATH to this nvidia folder, but then cupy cannot find the .dlls in the subfolders, so I copied the .dlls from the nvidia/subfolder/bin's to a an empty nvidia/bin folder i created. I didnt set any other PATHs than the one to site-packages. 

This works and I can run e.g. cp.random, cp.linalg.inv which was my use case
